### ISSUE TITLE
[Bug]: hybrid search with invalid offset expected to return an error but succeeded

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#32174]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20240411-a938951e-amd64
- Deployment mode(standalone or cluster): standalone
- MQ type(rocksmq, pulsar or kafka): rocksmq   
- SDK version(e.g. pymilvus v2.0.0rc2): go-sdk master branch
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

hybrid search with invalid offset: -1 and 16384+1, expected to return error but succ

- case:
```
func TestHybridSearchMultiVectorsPagination(t *testing.T) {
	//t.Skip("https://github.com/milvus-io/milvus-sdk-go/issues/693")
	t.Parallel()
	ctx := createContext(t, time.Second*common.DefaultTimeout*2)
	// connect
	mc := createMilvusClient(ctx, t)

	// create -> insert [0, 3000) -> flush -> index -> load
	cp := CollectionParams{CollectionFieldsType: AllVectors, AutoID: false, EnableDynamicField: false,
		ShardsNum: common.DefaultShards, Dim: common.DefaultDim}

	dp := DataParams{DoInsert: true, CollectionFieldsType: AllVectors, start: 0, nb: common.DefaultNb,
		dim: common.DefaultDim, EnableDynamicField: false}

	// index params
	ips := GenDefaultIndexParamsForAllVectors()
	collName := prepareCollection(ctx, t, mc, cp, WithDataParams(dp), WithIndexParams(ips), WithCreateOption(client.WithConsistencyLevel(entity.ClStrong)))

	// hybrid search with different limit
	sp, _ := entity.NewIndexFlatSearchParam()
	expr := fmt.Sprintf("%s > 4", common.DefaultIntFieldName)
	queryVec1 := common.GenSearchVectors(1, common.DefaultDim, entity.FieldTypeFloatVector)
	queryVec2 := common.GenSearchVectors(1, common.DefaultDim, entity.FieldTypeFloat16Vector)
	for _, invalidOffset := range []int64{-1, common.MaxTopK + 1} {
		sReqs := []*client.ANNSearchRequest{
			client.NewANNSearchRequest(common.DefaultFloatVecFieldName, entity.L2, "", queryVec1, sp, common.DefaultTopK, client.WithOffset(invalidOffset)),
			client.NewANNSearchRequest(common.DefaultFloat16VecFieldName, entity.L2, "", queryVec2, sp, common.DefaultTopK),
		}
		_, errSearch := mc.HybridSearch(ctx, collName, []string{}, common.DefaultTopK, []string{}, client.NewRRFReranker(), sReqs)
		common.CheckErr(t, errSearch, false, "top k should be in range [1, 16384]")
	}

	// search with different reranker and offset
	for _, reranker := range []client.Reranker{
		client.NewRRFReranker(),
		client.NewWeightedReranker([]float64{0.8, 0.2}),
		client.NewWeightedReranker([]float64{0.0, 0.2}),
		client.NewWeightedReranker([]float64{0.4, 1.0}),
	} {
		sReqs := []*client.ANNSearchRequest{
			client.NewANNSearchRequest(common.DefaultFloatVecFieldName, entity.L2, expr, queryVec1, sp, common.DefaultTopK, client.WithOffset(5)),
			client.NewANNSearchRequest(common.DefaultFloat16VecFieldName, entity.L2, expr, queryVec2, sp, common.DefaultTopK),
		}
		// hybrid search
		searchRes, errSearch := mc.HybridSearch(ctx, collName, []string{}, common.DefaultTopK, []string{}, reranker, sReqs)
		common.CheckErr(t, errSearch, true)
		common.CheckSearchResult(t, searchRes, 1, common.DefaultTopK)
	}
}
```

- test result
![image](https://github.com/milvus-io/milvus/assets/27288593/025f1c7d-e77c-4e2c-9c97-2d91b9a904fd)

### Milvus Log

collection_name: bgRW
pods:
```
zong-go-master-2-etcd-0                                           1/1     Running                           0                14d     10.104.24.130   4am-node29   <none>           <none>
zong-go-master-2-kafka-0                                          1/1     Running                           0                14d     10.104.28.20    4am-node33   <none>           <none>
zong-go-master-2-kafka-1                                          1/1     Running                           0                14d     10.104.32.108   4am-node39   <none>           <none>
zong-go-master-2-kafka-2                                          1/1     Running                           0                14d     10.104.15.203   4am-node20   <none>           <none>
zong-go-master-2-kafka-zookeeper-0                                1/1     Running                           0                14d     10.104.28.21    4am-node33   <none>           <none>
zong-go-master-2-kafka-zookeeper-1                                1/1     Running                           0                14d     10.104.30.192   4am-node38   <none>           <none>
zong-go-master-2-kafka-zookeeper-2                                1/1     Running                           0                14d     10.104.23.27    4am-node27   <none>           <none>
zong-go-master-2-milvus-standalone-5d9db6b9d9-2n2nj               1/1     Running                           0                9h      10.104.6.76     4am-node13   <none>           <none>
zong-go-master-2-minio-669b58dcd8-gw7zz                           1/1     Running                           0                14d     10.104.30.193   4am-node38   <none>           <none>
```

**Comment by czs007** [comment]
The current implementation disregards the "offset" parameter in sub-route queries and instead relies on the outer "offset" parameter.

**Comment by ThreadDao** [comment]
@czs007 
Hybrid search with invalid outer "offset" succeeded. expected error
- image: milvus-io-master-02f17b8-20240417
- SDK: go-sdk master branch
- case:
```
ctx := createContext(t, time.Second*common.DefaultTimeout*2)
	// connect
	mc := createMilvusClient(ctx, t)

	// create -> insert [0, 3000) -> flush -> index -> load
	cp := CollectionParams{CollectionFieldsType: AllVectors, AutoID: false, EnableDynamicField: false,
		ShardsNum: common.DefaultShards, Dim: common.DefaultDim}

	dp := DataParams{DoInsert: true, CollectionFieldsType: AllVectors, start: 0, nb: common.DefaultNb,
		dim: common.DefaultDim, EnableDynamicField: false}

	// index params
	ips := GenDefaultIndexParamsForAllVectors()
	collName := prepareCollection(ctx, t, mc, cp, WithDataParams(dp), WithIndexParams(ips), WithCreateOption(client.WithConsistencyLevel(entity.ClStrong)))

	// hybrid search with different limit
	sp, _ := entity.NewIndexFlatSearchParam()
	expr := fmt.Sprintf("%s > 4", common.DefaultIntFieldName)
	queryVec1 := common.GenSearchVectors(1, common.DefaultDim, entity.FieldTypeFloatVector)
	queryVec2 := common.GenSearchVectors(1, common.DefaultDim, entity.FieldTypeFloat16Vector)
	// milvus ignore invalid offset with ANNSearchRequest
	for _, invalidOffset := range []int64{-20, common.MaxTopK + 1} {
		sReqs := []*client.ANNSearchRequest{
			client.NewANNSearchRequest(common.DefaultFloatVecFieldName, entity.L2, "", queryVec1, sp, common.DefaultTopK, client.WithOffset(invalidOffset)),
			client.NewANNSearchRequest(common.DefaultFloat16VecFieldName, entity.L2, "", queryVec2, sp, common.DefaultTopK),
		}
		_, errSearch := mc.HybridSearch(ctx, collName, []string{}, common.DefaultTopK, []string{}, client.NewRRFReranker(), sReqs)
		common.CheckErr(t, errSearch, true)

		//hybrid search with invalid offset
		_, errSearch = mc.HybridSearch(ctx, collName, []string{}, common.DefaultTopK, []string{}, client.NewRRFReranker(), sReqs, client.WithOffset(invalidOffset))
		common.CheckErr(t, errSearch, false, "top k should be in range [1, 16384]")
	}
```

**Comment by ThreadDao** [comment]
@binbinlv please add related case with pymilvus

**Comment by ThreadDao** [comment]
fixed master-20240425-2586c2f1-amd64
